File: modules/01-basic-syntax/typescript/scopes_shadowing.ts
// Scopes and shadowing
console.log('------------------------\n');

let a = 10;
console.log(`before: ${a}`);

{
  let a = `hello`;
  console.log(`inner scope: ${a}`);

  let a = true;
  console.log(`shadowed in inner scope: ${a}`);
}

console.log(`after: ${a}`);

File: modules/01-basic-syntax/python/scopes_shadowing.py
# Scopes and shadowing

print("------------------------\n")

a = 10
print(f"before: {a}")

with open("scopes_shadowing.py", "r") as this_file:
  a = "hello"
  print(f"inner scope: {a}")

  a = True
  print(f"shadowed in inner scope: {a}")

print(f"after: {a}")
